### ISSUE TITLE
Please be better than the default devtools

### ISSUE THREAD

**Issue by kumarharsh** [firebug/firebug.next#326]
This is more of a cry of desperation than a formal bug request, but perhaps something would come out of it...

As it stands, the firefox developer tools is a very shoddy cousin of Firebug, and the direction you guys have taken to integrate firebug into the default dev-tools can be a bad sign for some developers in the short run. I know that things would get better *eventually*, but that eventuality I feel is too far... and it's very ugly world till then.

Here are a bunch of issues I see which make it incredibly painful to work with devtools:

1. Incredibly poor Network tab: Apart from the very useful "Edit and Resend" functionality offered by the default devtools, the rest of the network tab is quite bad.
  a) You can't see the full URL at a glance,
  b) You can't see the response as a simple JSON. What's the reason for forcefully rendering the object as a expandable things is beyond me.
  c) Bad response parsing... [See Bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1066519)
  d) Wrong detection of those response objects: Arrays get shown as Objects, Empty objects don't expand, strings are cut short, and what not.
  e) No way to copy the response. -_-
  f) [See more](https://bugzilla.mozilla.org/show_bug.cgi?id=925340)

2. CSS Panel is broken:
  a) It is nigh-impossible to add a new style definition for pseudo-selectors. Once you right click on the CSS panel in the Inspector panel, and click on Add Rule, and then write something like `body:before` and press tab, poof, the thing is gone. Never to come back. The only way to do something like this is to go to the Style Editor panel. 
  b) Say you're typing something in the CSS panel like a long style for text-shadow, and press Shift+Home to select till the start of the string, it goes and inserts a stupid autocomplete entry at the start. Same with Shift+End. [See Bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1115638)

3. Console
  a) No way to select an object and use it in console. I mean, isn't that a pretty basic thing FB1.x was already offering?

Firebug 2.0 has it all figured out. Please consider these as some things you would need to work on so that these functionality bugs creep into FB3.0.

**Comment by thorinmesser** [comment]
Kind of wish github had a way to vote for issues right now. Spot on.

**Comment by fflorent** [comment]
Please, be aware that: 
* Reporting issues would help us (if you see they haven't been reported here)
* We haven't finished the work yet, progresses are expected
* We're free and open-source if you want to help us improve Firebug 3.0 as well as the Firefox DevTools ;)

Also please, if you want to discuss some points, make them in issues (if they are related to platform bugs, there can be a Firebug-side bug to track them).

Florent

**Comment by fflorent** [comment]
Closing is a mystake, sorry. We should consider discussing a bit more on these points and report issues according to them.

Florent

**Comment by kumarharsh** [comment]
I have already made bug reports for most of the things I've discussed above. Some of them, actually, the most irritating bugs, have reports dating back about an year, or perhaps more, with no activity at all. I can't help but imagine that Dev Tools is not a priority for the Firefox development team. But this is Firebug, this is the only priority here, so I'm hoping these issues would be taken more seriously.

Regards
Harsh

**Comment by kumarharsh** [comment]
@janodvarko @fflorent  I gave the Firebug 3.0 alpha 14 a try today, and it hurts to say that **all** of the things I pointed out above still exist. Right now, Firebug 3.0 seems like just a paint-job over the dev-tools, and that too (excuse my saying so) not a pretty one.

![screenshot from 2015-10-26 17 22 51](https://cloud.githubusercontent.com/assets/760112/10728301/a806f5a6-7c06-11e5-913e-24d54db00a99.png)
![screenshot from 2015-10-26 17 23 12](https://cloud.githubusercontent.com/assets/760112/10728302/a808ca16-7c06-11e5-85e5-7737e8decb3c.png)

Frankly, I don't care about the prettiness, just start getting those features back into FF. I've been waiting for the built-in dev-tools to get better in futility, while Chrome is just blazing forward at an incredible speed adding incredibly powerful tools for developers with every few releases (sometimes at the expense of the general public :smirk:). I don't know whom Firefox is targeting anymore, they can't get even their absolutely *basic* dev-tools straight.
